test.js

```diff
@@ -0,0 +1,36 @@
+const test = require('tape')
+const noNodeConfig = require('./index')
+const co = require('co')
+const config = require('./testConfig/configData')
+const keys = ['mongo', 'redis']
+const PROCESS_ENV = process.env.NODE_ENV = 'test_env'
+
+const configObject = co(function* () {
+    return yield noNodeConfig({config})
+})
+
+test('Should initialize config object', (t) => { 
+    t.plan(3)
+    configObject.then((conf) => {
```

Comment:
@anuragCES here you use `Promises` directly, but code would be much nicer if you'd use generators + `yield` instead. To do this use generator function wrapped by `co` for `test()` function callback:
```js
test('Should initialize config object', co.wrap(function* (t) {
    try {
        let result = yield configObject
        // assertions
    } catch (e) {
        t.equal(e, undefined, 'No error')
    }
    t.end()
})) 
```

In the example above notice the try-catch, this helps to cover a case when `configObject` Promise got rejected, something you haven't covered. I assume in your implementation if this would happen, test code will hang (since no assertions inside `.then()` would be executed). This also shows why it's better to use `t.end()` instead of `t.plan()` (number of assertions would vary depending of `configObject` result)